compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -2935,13 +2935,15 @@ impl<'tcx> InferCtxtPrivExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
             rustc_transmute::Assume::from_const(self.infcx.tcx, obligation.param_env, trait_ref.substs.const_at(3)) else {
                 span_bug!(span, "Unable to construct rustc_transmute::Assume where it was previously possible");
             };
+        // FIXME(bryangarza): Is this enough, or should we resolve all nested
+        // obligations like we do for `confirm_transmutability_candidate(...)?`
```

Comment:
There shouldn't be any circumstances under which this is producing nested obligations. If we're here, it's because the result we got back was an `Err`.

Reply:
But couldn't the original call to confirm_transmutability_candidate have had the Err somewhere in the nested obligations? Remember, we are checking transmutability from scratch again here, so if we get an Answer tree, it's not flattened yet

Reply:
I'll remove this for now and if we seem to run into issues we can just address it, otherwise the comment will just be confusing for future readers...